src/pages/files.tsx

```diff
@@ -1,64 +1,383 @@
+          const finalContent = isImage ? base64 : atob(base64);
+          const finalFileName = file.name;
+
+          try {
+            await props.contentsManager.save(finalFileName, {
+              type: 'file',
+              format: isImage ? 'base64' : 'text',
+              content: finalContent
+            });
+
+            uploadedFiles.push({
+              id: fileId,
+              name: finalFileName,
+              size: file.size,
+              type: file.type,
+              lastModified: file.lastModified,
+              content
+            });
+          } catch (error) {
+            console.warn(`Upload skipped or failed for ${finalFileName}`, error);
+          }
+        }
+
+        if (uploadedFiles.length > 0) {
+          props.onFilesUploaded(uploadedFiles);
+        }
+      } finally {
+        props.onUploadEnd();
+      }
+    },
+    [props]
+  );
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      handleFileSelect(e.target.files);
+    }
+  };
+
+  const triggerFileSelect = () => {
+    fileInputRef.current?.click();
+  };
+
+  // Expose the trigger function to the parent
+  React.useImperativeHandle(ref, () => ({
+    triggerFileSelect
+  }));
+
   return (
-    <button className="je-Tile">
+    <input
+      ref={fileInputRef}
+      type="file"
+      multiple
+      onChange={handleInputChange}
+      style={{ display: 'none' }}
+      accept=".png,.jpg,.jpeg,.csv,image/png,image/jpeg,text/csv"
+    />
+  );
+});
+
+FileUploader.displayName = 'FileUploader';
+
+/**
+ * This component displays a thumbnail for each uploaded image or file.
+ * We use the `getFileIcon` function to determine the icon based on file type.
+ */
+interface IFileThumbnailProps {
+  file: IUploadedFile;
+  onRemove: (fileId: string) => void;
+  contentsManager: Contents.IManager;
+}
+
+/**
+ * FileThumbnail component displays a thumbnail for an uploaded file.
+ * @param props - The properties for the FileThumbnail component, including the file to display and a callback for removing the file.
+ * @returns A JSX element representing the file thumbnail.
+ */
+function FileThumbnail(props: IFileThumbnailProps) {
+  const { file } = props;
+
+  const handleRemove = async () => {
+    try {
+      await props.contentsManager.delete(file.name);
+    } catch (err) {
+      console.error('Error removing file:', err);
+    } finally {
+      props.onRemove(file.id);
+    }
+  };
+
+  const fileIcon = getFileIcon(file.name, file.type);
+  return (
+    <div className="je-FileThumbnail">
+      <div className="je-FileThumbnail-preview">
+        <div className="je-FileThumbnail-icon">
+          <fileIcon.react />
+        </div>
+      </div>
+      <div className="je-FileThumbnail-info">
+        <div className="je-FileThumbnail-name" title={file.name}>
+          {file.name}
+        </div>
+        <div className="je-FileThumbnail-size">{(file.size / 1024).toFixed(1)} KB</div>
```

Comment:
I don't think this was on the mockups?

Reply:
I think it makes it take up more space than on figma. I personally think that this is looks nice, but wanted to make sure we checked with @ivonnem3

![image](https://github.com/user-attachments/assets/655fec5a-9210-4634-8028-9315a5bb85b3)

Alternatives include showing this in title on hover or not showing it at all.

Apologies if this was already discussed and I am just adding noise.